src/cascadia/TerminalSettingsEditor/KeyChordListener.cpp

```diff
@@ -0,0 +1,147 @@
+
+            if (isDown)
+            {
+                switch (mod)
+                {
+                case VirtualKey::Control:
+                case VirtualKey::LeftControl:
+                case VirtualKey::RightControl:
+                {
+                    flags |= VirtualKeyModifiers::Control;
+                    break;
+                }
+                case VirtualKey::Menu:
+                case VirtualKey::LeftMenu:
+                case VirtualKey::RightMenu:
+                {
+                    flags |= VirtualKeyModifiers::Menu;
+                    break;
+                }
+                case VirtualKey::Shift:
+                case VirtualKey::LeftShift:
+                {
+                    flags |= VirtualKeyModifiers::Shift;
+                    break;
+                }
+                case VirtualKey::LeftWindows:
+                case VirtualKey::RightWindows:
+                {
+                    flags |= VirtualKeyModifiers::Windows;
+                    break;
+                }
+                }
+            }
+        }
+        return flags;
+    }
+
+    KeyChordListener::KeyChordListener()
+    {
+        InitializeComponent();
+        _InitializeProperties();
+
+        PropertyChanged([this](auto&&, const PropertyChangedEventArgs& args) {
+            if (args.PropertyName() == L"ProposedKeys")
+            {
+                KeyChordTextBox().Text(Model::KeyChordSerialization::ToString(_ProposedKeys));
+            }
+        });
+    }
+
+    void KeyChordListener::_InitializeProperties()
+    {
+        // Initialize any KeyChordListener dependency properties here.
+        // This performs a lazy load on these properties, instead of
+        // initializing them when the DLL loads.
+        if (!_CurrentKeysProperty)
+        {
+            _CurrentKeysProperty =
+                DependencyProperty::Register(
+                    L"CurrentKeys",
+                    xaml_typename<Control::KeyChord>(),
+                    xaml_typename<Editor::KeyChordListener>(),
+                    PropertyMetadata{ nullptr, PropertyChangedCallback{ &KeyChordListener::_OnCurrentKeysChanged } });
+        }
+    }
+
+    void KeyChordListener::_OnCurrentKeysChanged(DependencyObject const& d, DependencyPropertyChangedEventArgs const& e)
+    {
+        if (auto control{ d.try_as<Editor::KeyChordListener>() })
+        {
+            auto controlImpl{ get_self<KeyChordListener>(control) };
+            controlImpl->ProposedKeys(unbox_value<Control::KeyChord>(e.NewValue()));
+        }
+    }
+
+    void KeyChordListener::KeyChordTextBox_LosingFocus(IInspectable const& /*sender*/, LosingFocusEventArgs const& /*e*/)
+    {
+        // export our key chord into the attached view model
+        SetValue(_CurrentKeysProperty, _ProposedKeys);
+    }
+
+    void KeyChordListener::KeyChordTextBox_KeyDown(IInspectable const& /*sender*/, KeyRoutedEventArgs const& e)
+    {
```

Comment:
does this need to do some accessibility notifying ala the command palette with RaiseNotificationEvent?

Reply:
i.e. how does it sound to be editing a key as a narrator user...

Reply:
Whoops! Went ahead and added...
- auto props for the "set action" combo box
- auto props for the key chord listener
- a notification event for when the text is changed

Now it works as it should...
- notify a user when a new key chord is bound
- tell the user what this control is (and how to use it)
- tell the user what they're modifying